astropy/timeseries/periodograms/lombscargle/implementations/utils.py

```diff
@@ -79,7 +102,273 @@ def extirpolate(x, y, N=None, M=4):
     return result
 
 
-def trig_sum(t, h, df, N, f0=0, freq_factor=1, oversampling=5, use_fft=True, Mfft=4):
+def jn(n, x, num_terms=10):
+    """
+    Approximate the Bessel function of the first kind
+    of an integer order n for small values of x.
+    Approximately equivalent to scipy.special.jv(n, x)
+    """
+    n = np.asarray(n, dtype=np.int32)
+    x = np.asarray(x)
+    half_x = 0.5 * x
+
+    n = n.astype(np.int64)
+
+    # Keep the array longer to approximately ignore the factorials of negative integers in the denominator
+    n_max = int(np.max(n)) + num_terms + 10
+    fact = np.empty(n_max + 1, dtype=np.float64)
+    fact[0] = 1.0
+    for i in range(1, n_max + 1):
+        fact[i] = fact[i - 1] * i
```

Comment:
I can't help feeling that there should be a more idiomatic to write this in Python, but I cannot find one that feels as readable. Oh well.